codec-http2/src/test/java/io/netty/handler/codec/http2/HpackDecoderTest.java

```diff
@@ -198,68 +221,102 @@ public void testLiteralHuffmanEncodedWithEmptyNameAndValue() throws Http2Excepti
         }
     }
 
-    @Test(expected = Http2Exception.class)
+    @Test
     public void testLiteralHuffmanEncodedWithPaddingGreaterThan7Throws() throws Http2Exception {
         byte[] input = {0, (byte) 0x81, -1};
-        ByteBuf in = Unpooled.wrappedBuffer(input);
+        final ByteBuf in = Unpooled.wrappedBuffer(input);
         try {
-            hpackDecoder.decode(0, in, mockHeaders, true);
+            assertThrows(Http2Exception.class, new Executable() {
+                @Override
+                public void execute() throws Throwable {
+                    hpackDecoder.decode(0, in, mockHeaders, true);
+                }
+            });
         } finally {
             in.release();
         }
     }
 
-    @Test(expected = Http2Exception.class)
+    @Test
     public void testLiteralHuffmanEncodedWithDecodingEOSThrows() throws Http2Exception {
         byte[] input = {0, (byte) 0x84, (byte) 0xFF, (byte) 0xFF, (byte) 0xFF, (byte) 0xFF};
-        ByteBuf in = Unpooled.wrappedBuffer(input);
+        final ByteBuf in = Unpooled.wrappedBuffer(input);
         try {
-            hpackDecoder.decode(0, in, mockHeaders, true);
+            assertThrows(Http2Exception.class, new Executable() {
+                @Override
+                public void execute() throws Throwable {
+                    hpackDecoder.decode(0, in, mockHeaders, true);
+                }
+            });
         } finally {
             in.release();
         }
     }
 
-    @Test(expected = Http2Exception.class)
+    @Test
     public void testLiteralHuffmanEncodedWithPaddingNotCorrespondingToMSBThrows() throws Http2Exception {
         byte[] input = {0, (byte) 0x81, 0};
-        ByteBuf in = Unpooled.wrappedBuffer(input);
+        final ByteBuf in = Unpooled.wrappedBuffer(input);
         try {
-            hpackDecoder.decode(0, in, mockHeaders, true);
+            assertThrows(Http2Exception.class, new Executable() {
+                @Override
+                public void execute() throws Throwable {
+                    hpackDecoder.decode(0, in, mockHeaders, true);
+                }
+            });
         } finally {
             in.release();
         }
     }
 
-    @Test(expected = Http2Exception.class)
+    @Test
     public void testIncompleteIndex() throws Http2Exception {
         byte[] compressed = StringUtil.decodeHexDump("FFF0");
-        ByteBuf in = Unpooled.wrappedBuffer(compressed);
+        final ByteBuf in = Unpooled.wrappedBuffer(compressed);
         try {
-            hpackDecoder.decode(0, in, mockHeaders, true);
-            assertEquals(1, in.readableBytes());
-            hpackDecoder.decode(0, in, mockHeaders, true);
```

Comment:
I wonder if the second call was the one that was supposed to throw, but the old test clearly couldn't tell the difference.

Reply:
when I worked on this the first one was actually throwing... Thats why I rewrote it.